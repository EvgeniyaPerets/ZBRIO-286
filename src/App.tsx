import { Button, Checkbox, Container, IconButton, List, ListItem, ListItemSecondaryAction, ListItemText, TextField, Typography } from '@material-ui/core';
import DeleteIcon from "@material-ui/icons/Delete";
import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { RootState, AppDispatch } from './redux/store'
import { addTodo, removeTodo, setTodoStatus } from './redux/todoSlice';

const App = () => {
  const [todoDescription, setTodoDescription ] = useState("");

  const todoList = useSelector((state: RootState) => state);
  const dispatch = useDispatch<AppDispatch>();

  return (
    <Container maxWidth="xs">
      <Typography style={{ textAlign: 'center' }} variant='h3'>
        Redux LIst App
      </Typography>
      <TextField
        variant='outlined'
        label='to od them'
        fullWidth
        onChange={(e) => setTodoDescription(e.target.value)}
        value={todoDescription}
      />
      <Button
        variant='contained'
        color='primary'
        fullWidth
        onClick={() => {
          dispatch(addTodo(todoDescription));
          setTodoDescription('')
        }}
      >
        Add Item
      </Button>
      <List>
        {todoList.map((todo) => (
          <ListItem key={todo.id}>
            <ListItemText
              style={{
                textDecoration: todo.completed ? 'line-through' : 'none'
              }}
            >
              {todo.discription}
            </ListItemText>
            <ListItemSecondaryAction>
              <IconButton
                onClick={() => {
                  dispatch(removeTodo(todo.id));
                }}
              >
                <DeleteIcon />
              </IconButton>
              <Checkbox
                edge='end'
                value={todo.completed}
                onChange={() => {
                  dispatch(setTodoStatus({ completed: !todo.completed, id: todo.id }))
                }}
              />
            </ListItemSecondaryAction>
          </ListItem>
        ))}
      </List>
    </Container>
  );
}

export default App;
