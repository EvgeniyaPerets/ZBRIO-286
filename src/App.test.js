import { fireEvent, render, screen } from "@testing-library/react"
import ProviderWrapper from "./ProviderWrapper";
import '@testing-library/jest-dom'
import App from "./App";
import userEvent from "@testing-library/user-event";

describe('test app', () => {
  it('app render correctly', () =>{
    const { getByText, asFragment, queryByText } = render(<ProviderWrapper />);
    const btn = getByText(/Add Item/i);
    const btn1 = queryByText(/edit item/i)

    expect(btn).toBeInTheDocument()
    expect(btn1).toBeNull()
    expect(asFragment(<App />)).toMatchSnapshot()
  })

  it('screen', () => {
    render(<ProviderWrapper />)
    screen.debug()
  })

  it('add item text', () => {
    const { getByRole, asFragment } = render(<ProviderWrapper />);
    const input = getByRole('textbox')
    const btn = getByRole('button')

    // userEvent.type(input, 'gdjevihb')
    fireEvent.change(input, { target: { value : 'gdjevihb'}})
    expect(input).toHaveValue('gdjevihb')

    fireEvent.click(btn)

    // expect(asFragment(<App />)).toMatchSnapshot()

    const list = getByRole('list')
    expect(list).toBeInTheDocument

    const checkbox = getByRole('checkbox')
    expect(checkbox).not.toBeChecked()
    fireEvent.click(checkbox)
    expect(checkbox).toBeChecked()

  })
})