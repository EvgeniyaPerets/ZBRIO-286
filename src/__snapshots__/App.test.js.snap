// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test app add item text 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3"
      style="text-align: center;"
    >
      Redux LIst App
    </h3>
    <div
      class="MuiBox-root MuiBox-root-11"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused"
          data-shrink="true"
        >
          to do them
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-14 PrivateNotchedOutline-legendNotched-15"
            >
              <span>
                to do them
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Add Item
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <ul
      class="MuiList-root MuiList-padding"
    >
      <li
        class="MuiListItem-container"
      >
        <div
          class="MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction"
        >
          <div
            class="MuiListItemText-root"
            style="text-decoration: none;"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              gdjevihb
            </span>
          </div>
        </div>
        <div
          class="MuiListItemSecondaryAction-root"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-19"
                data-indeterminate="false"
                type="checkbox"
                value="false"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`test app app render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3"
      style="text-align: center;"
    >
      Redux LIst App
    </h3>
    <div
      class="MuiBox-root MuiBox-root-1"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          to do them
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-4"
            >
              <span>
                to do them
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Add Item
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <ul
      class="MuiList-root MuiList-padding"
    />
  </div>
</DocumentFragment>
`;
